contracts/src/v0.8/dev/interfaces/OCR2DRRegistryInterface.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+/**
+ * @title OCR2DR billing interface.
+ */
+interface OCR2DRRegistryInterface {
+  struct RequestCommitment {
+    uint64 blockNum;
+    uint64 subId;
+    uint32 callbackGasLimit;
+    bytes data;
+    address sender;
+  }
+
+  /**
+   * @notice Get configuration relevant for making requests
+   * @return minimumRequestConfirmations global min for request confirmations
+   * @return maxGasLimit global max for request gas limit
+   * @return s_provingKeyHashes list of registered key hashes
+   */
+  function getRequestConfig()
+    external
+    view
+    returns (
+      uint16,
+      uint32,
+      bytes32[] memory
+    );
+
+  /**
+   * @notice Initiate the billing process for an OCR2DR request to the DON
+   * @param keyHash - Corresponds to a particular oracle job which uses
+   * that key for generating the OCR2DR proof. Different keyHash's have different gas price
+   * ceilings, so you can select a specific one to bound your maximum per request cost.
+   * @param subId  - The ID of the OCR2DR subscription. Must be funded
+   * with the minimum subscription balance required for the selected keyHash.
+   * @param minimumRequestConfirmations - How many blocks you'd like the
+   * oracle to wait before responding to the request. See SECURITY CONSIDERATIONS
+   * for why you may want to request more. The acceptable range is
+   * [minimumRequestBlockConfirmations, 200].
+   * @param callbackGasLimit - How much gas you'd like to receive in your
+   * fulfillRequest callback. Note that gasleft() inside fulfillRequest
+   * may be slightly less than this amount because of gas used calling the function
+   * (argument decoding etc.), so you may need to request slightly more than you expect
+   * to have inside fulfillRequest. The acceptable range is
+   * [0, maxGasLimit]
+   * @param data Encoded OCR2DR request data, use OCR2DRClient API to encode a request
+   * @return requestId - A unique identifier of the request. Can be used to match
+   * a request to a response in fulfillRequest.
+   */
+  function sendRequest(
+    bytes32 keyHash,
+    uint64 subId,
+    uint16 minimumRequestConfirmations,
+    uint32 callbackGasLimit,
+    bytes calldata data
+  ) external returns (bytes32);
+
+  /*
+   * @notice Conclude billing for an OCR2DR request
+   * @param proof contains the proof and response
+   * @param rc request commitment pre-image, committed to at request time
+   * @return payment amount billed to the subscription
+   * @dev simulated offchain to determine if sufficient balance is present to fulfill the request
+   */
+  function fulfillRequest(
+    bytes32 requestId,
+    bytes32 keyHash,
+    RequestCommitment memory rc,
```

Comment:
We could strip this out of being an input parameter, but then the gas configurations for the request would need to be stored within the registry to be looked up during fulfillment.

Reply:
If it isn't too difficult, I would prefer we strip this out now to reduce gas usage & complexity.